src/imitation/util/util.py

```diff
@@ -135,3 +135,28 @@ def helper(obj):
         return obj
 
     return helper
+
+
+def endless_iter(iterable: Iterable) -> Generator:
```

Comment:
IIRC this is `itertools.chain.from_iterable(itertools.repeat(iterable))`.

Reply:
Oh nice. Although not quite the same in that `endless_iter` will sanity check against repeating an iterator (which would lead to hanging).

Reply:
Since you have an `iterable` (not an iterator) I think you can do something like:

```python
try:
    next(iter(iterable)))
except StopIteration:
    raise RuntimeError(f"iterable {iterable} had no elements to iterate over.")
return itertools.chain.from_iterable(itertools.repeat(iterable))
```

Reply:
The current implementation also accounts for a nondeterministically-lengthed `__iter__` which could be empty on the 3rd call etc. But then again, if it were empty only sometimes, then we wouldn't run into the "hanging on infinite empty Iterables" problem.

I'll probably replace this with @AdamGleave's solution since it's a bit simpler. will think about this a bit more.

Reply:
I ended up coding Adam's solution except using a `yield from` instead of a `return`. Because Generator has a `.close()` method, but `Iterator` doesn't. I'm questioning this decision because it's not clear to me right now that `.close()` actually adds any value for its complexity right now.

Reply:
Finally decided to use `return` and get rid of all the `.close()` related clean ups, because they didn't seem to be necessary for this iterator. Will add them back in if when working with MineRL, our DataLoader seems to need this clean up explicitly.

Reply:
I guess the argument for `close` would be if it depended on resources that need to be explicitly cleared up (will not be garbage collected): e.g. if you're computing rollouts from an expert policy, you might need to explicitly destroy the policy when no longer needed. My impression is PyTorch does a better job of being GC-compatible than Tensorflow though so this may not be needed.